### PR TITLE
Allow default requests parameters like proxy to be defined in extra options field of a Airflow HTTP Connection

### PR DESCRIPTION
**Pull request by dabla**
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by Taragolis** [comment]
Also required to update the documentation for the connection:

https://github.com/apache/airflow/blob/d7bc217957b65471ca5f2e259bba15c71a2b0c41/docs/apache-airflow-providers-http/connections/http.rst?plain=1#L56-L57

**Comment by dabla** [comment]
> Also required to update the documentation for the connection:
> 
> https://github.com/apache/airflow/blob/d7bc217957b65471ca5f2e259bba15c71a2b0c41/docs/apache-airflow-providers-http/connections/http.rst?plain=1#L56-L57

I've updated the documentation.

**Review by dirrao — CHANGES_REQUESTED** [review]
I see some checks are failing. Can you check and fix them? 
I would suggest to install pre-commit hook for static checks.
https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&ved=2ahUKEwi4lubf1tWDAxWISWwGHdq7APoQFnoECA4QAQ&url=https%3A%2F%2Fgithub.com%2Fapache%2Fairflow%2Fblob%2Fmain%2FSTATIC_CODE_CHECKS.rst&usg=AOvVaw3_chGWjXp6V2QRaCTm_fyE&opi=89978449

File: tests/providers/http/hooks/test_http.py [hunk 1]
```diff
@@ -525,3 +588,111 @@ async def test_async_request_uses_connection_extra(self, aioresponse):
...
+        assert actual == {"bearer": "test"}
+
+    def test_process_extra_options_from_connection_when_max_redirects_is_defined(self):
+        extra_options = {}
+        conn = get_airflow_connection_with_extra(extra={"bearer": "test", "max_redirects": 3})()
+
+        actual = HttpAsyncHook._process_extra_options_from_connection(conn=conn, extra_options=extra_options)
+
+        assert extra_options == {"max_redirects": 3}
+        assert actual == {"bearer": "test"}
```
Comment (aritra24) [inline tests/providers/http/hooks/test_http.py]: This all looks like it's testing the same thing with different params, can this perhaps just all be in one test?

Comment (dabla) [reply tests/providers/http/hooks/test_http.py]: This has been fixed and refactored as one test now

**Comment by jscheffl** [comment]
Just a small type checking glitch, I assume then we are ready to merge :-D

**Comment by dabla** [comment]
> Just a small type checking glitch, I assume then we are ready to merge :-D

Fixed it

**Review by Taragolis — APPROVED** [review]

**Review by jscheffl — APPROVED** [review]

**Pull request merged**